contracts/contracts/RaidenMicroTransferChannels.sol

```diff
@@ -295,15 +297,72 @@ contract RaidenMicroTransferChannels {
         // ! Note that EIP712 might change how hashing is done, triggering a
         // new contract deployment with updated code.
         bytes32 message_hash = keccak256(
-          keccak256('address receiver', 'uint32 block_created', 'uint192 balance', 'address contract'),
-          keccak256(_receiver_address, _open_block_number, _balance, address(this))
+            keccak256(
+                'string messageID',
+                'address receiver',
+                'uint32 block_created',
+                'uint192 balance',
+                'address contract'
+            ),
+            keccak256(
+                'Sender balance proof signature',
+                _receiver_address,
+                _open_block_number,
+                _balance,
+                address(this)
+            )
         );
 
         // Derive address from signature
         address signer = ECVerify.ecverify(message_hash, _balance_msg_sig);
         return signer;
     }
 
+    /// @dev Returns the receiver address extracted from the closing signature.
+    /// Works with eth_signTypedData https://github.com/ethereum/EIPs/pull/712.
+    /// @param _sender_address The address that sends tokens.
+    /// @param _open_block_number The block number at which a channel between the
+    /// sender and receiver was created.
+    /// @param _balance The amount of tokens owed by the sender to the receiver.
+    /// @param _closing_sig The receiver's signed balance message, containing the sender's address.
+    /// @return Address of the closing signature signer.
+    function verifyClosingSignature(
```

Comment:
The function name here is misleading. Either change it to `extractClosingSignature` or add the verification check that you perform [outside](https://github.com/raiden-network/microraiden/pull/273/files#diff-f70d66082794ad8ac4eb9c1e46e17371R196) of the function inside.

Reply:
The reason was name consistency with `verifyBalanceProof` which does the same thing.
So, in this case I should also change `verifyBalanceProof` to `extractBalanceProofSignature`, right?

Reply:
I would say so yes